crates/viewer/re_redap_browser/src/requested_object.rs

```diff
@@ -0,0 +1,101 @@
+use crossbeam_channel::{bounded, Receiver};
+
+use re_viewer_context::AsyncRuntimeHandle;
+
+/// A handle to an object that is requested asynchronously.
+pub enum RequestedObject<T: Send + 'static> {
+    Pending(Receiver<T>),
+    Completed(T),
+}
+
+impl<T: Send + 'static> RequestedObject<T> {
+    /// Create a new [`Self`] with the given future.
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
I've add to duplicate these function because native _wants_ `Send`, but wasm doesn't want it at all. I have no idea if there is a better pattern here.

Reply:
We should do like wgpu does and add this utility https://github.com/wumpf/wgpu/blob/d24dbc9b0ee54256a256287dc053c8e3cc5398a3/wgpu-types/src/lib.rs#L8026
at least I believe we can solve it with that

Reply:
worked 🎉